File: javascript/dsa/data-structures/array/move-zeroes.js
function moveZeroes(nums) {
  let lastNonZeroAt = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      let temp = nums[i];
      nums[i] = nums[lastNonZeroAt];
      nums[lastNonZeroAt] = temp;
      lastNonZeroAt++;
    }
  }
}

File: python/dsa/data-structures/array/move-zeroes.py
def moveZeroes(nums):
    last_non_zero_at = 0
    for i in range(len(nums)):
        if nums[i] != 0:
            temp = nums[i]
            nums[i] = nums[last_non_zero_at]
            nums[last_non_zero_at] = temp
            last_non_zero_at += 1
